grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/autoconfigure/metrics/GRpcMetricsAutoConfiguration.java

```diff
@@ -182,24 +193,29 @@ public ServerInterceptor measure(MeterRegistry registry, GRpcMetricsProperties m
 
     @Bean
     public GRpcMetricsTagsContributor defaultTagsContributor(GRpcServerProperties properties) {
-        final Boolean hasMultipleAddresses = Optional.ofNullable(properties.getNettyServer())
-                .map(GRpcServerProperties.NettyServerProperties::getAdditionalListenAddresses)
-                .map(l -> !l.isEmpty())
-                .orElse(false);
-
-        return (status, methodDescriptor, attributes) -> {
-
-            final ArrayList<Tag> tags = new ArrayList<>();
-            tags.add(Tag.of("result", status.getCode().name()));
-            tags.add(Tag.of("method", methodDescriptor.getFullMethodName()));
-            if (hasMultipleAddresses) {
-                Optional.ofNullable(attributes)
-                        .map(a -> a.get(Grpc.TRANSPORT_ATTR_LOCAL_ADDR))
-                        .map(SocketAddress::toString)
-                        .map(a -> Tag.of("address", a))
-                        .ifPresent(tags::add);
-            }
-            return tags;
-        };
+        final boolean hasMultipleAddresses = Optional.ofNullable(properties.getNettyServer())
+            .map(GRpcServerProperties.NettyServerProperties::getAdditionalListenAddresses)
+            .map(l -> !l.isEmpty())
+            .orElse(false);
+
+        GRpcMetricsTagsContributor defaultContributor = (status, methodDescriptor, attributes) -> Tags.of(
+            "result", status.getCode().name(),
+            "method", methodDescriptor.getFullMethodName()
+        );
+
+        if (hasMultipleAddresses) {
+            return (status, methodDescriptor, attributes) -> {
+                String address = Optional.ofNullable(attributes)
+                    .map(a -> a.get(Grpc.TRANSPORT_ATTR_LOCAL_ADDR))
+                    .map(SocketAddress::toString)
+                    .orElse("");
```

Comment:
Prometheus does not tolerate the same metric having different sets of tags, so `""` is a safer default than not adding the tag.